docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/kubernetes.md

```diff
@@ -113,14 +138,315 @@ helm --namespace dagster-cloud upgrade agent \
+
+<Tabs>
+<TabItem value="agent-secrets" label="All code locations">
+
+```yaml file=values.yaml
+# values.yaml
+workspace:
+  envSecrets:
+    - name: database-password
 ```
+
+```shell
+helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --values ./values.yaml
+```
+
+`envSecrets` will make the secret available in an environment variable, see the Kubernetes docs on [`envFrom` for details](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables). In this example the environment variable `DATABASE_PASSWORD` would have the value `your_password`.
+
+</TabItem>
+
+<TabItem value="code-location-secrets" label="Single code location">
+
+Modify the [`dagster_cloud.yaml` file](/todo) in your project's Git repository:
+
+```yaml file=dagster_cloud.yaml
+location:
+  - location_name: cloud-examples
+    image: dagster/dagster-cloud-examples:latest
+    code_source:
+      package_name: dagster_cloud_examples
+    container_context:
+      k8s:
+        env_secrets:
+          - database-password
+```
+
+`env_secrets` will make the secret available in an environment variable, see the Kubernetes docs on [`envFrom` for details](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables). In this example the environment variable `DATABASE_PASSWORD` would have the value `your_password`.
+
+
+</TabItem>
+</Tabs>
+
+:::note
+If you need to request secrets from a secret manager like AWS Secrets Manager or Hashicorp Vault, follow one of the prior methods to give your code access to vault credentials. Then, inside your Dagster code, use those  credentials to authenticate a Python client that requests secrets from the secret manager.
+:::
+
+
+### Use a different service account for a specific code location
+
+Modify the [`dagster_cloud.yaml` file](/todo) in your project's Git repository:
+
+```yaml file=dagster_cloud.yaml
+locations:
+  - location_name: cloud-examples
+    image: dagster/dagster-cloud-examples:latest
+    code_source:
+      package_name: dagster_cloud_examples
+    container_context:
+      k8s:
+        service_account_name: my_service_account_name
+```
+
+### Run Dagster+ with different Kubernetes clusters
+
+<Tabs>
+<TabItem value="ha" label="Work can happen in either cluster">
+
+Deploy the agent Helm chart to each cluster, setting the `isolatedAgents.enabled` flag to true.
+
+```yaml file=values.yaml
+# values.yaml
+isolatedAgents:
+  enabled: true
+```
+
+```shell
+helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --values ./values.yaml
+```
+
+In this configuration, requests will be randomly distributed across agents in both clusters.
+
+</TabItem>
+<TabItem value="queues" label="Global asset graph, but projects run in different clusters">
```

Comment:
not sure how to draw the line between this being a "how to do it on k8s" section in the agent queues docs vs. 'how to do agent queues' in the k8s docs - maybe having both is fine?